src/codecoverage/backend/codecoverage_backend/services/gcp.py

```diff
@@ -249,3 +271,44 @@ def get_coverage(self, repository, changeset, path):
         assert os.path.exists(report_path), 'Missing report {}'.format(report_path)
 
         return covdir.get_path_coverage(report_path, path)
+
+    def get_history(self, repository, path='', start=None, end=None):
+        '''
+        Load the history overall coverage from the redis cache
+        Default to date range from now back to a year
+        '''
+        if end is None:
+            end = calendar.timegm(datetime.utcnow().timetuple())
+        if start is None:
+            start = end - 365*24*3600
+        assert isinstance(start, int)
+        assert isinstance(end, int)
+        assert end > start
+
+        # Load changesets ordered by date, in that range
+        history = self.redis.zrevrangebyscore(
+            KEY_HISTORY.format(repository=repository),
+            end, start,
+            withscores=True,
+        )
+
+        def _coverage(changeset, date):
+            # Load overall coverage for specified path
+            changeset = changeset.decode('utf-8')
+            key = KEY_OVERALL_COVERAGE.format(
+                repository=repository,
+                changeset=changeset,
+            )
+            coverage = self.redis.hget(key, path)
```

Comment:
Will this be fast enough? We'll call hget thousand of times

Reply:
As discussed on IRC, we have several options to optimize later on:
1. Use redis pipelining to batch requests
2. Store data more efficiently for retrieval (`overall:<repo>:<path>` :arrow_right: `[0.1, 0.2, ...]`
3. Build dedicated reports on the bot

Reply:
Want to file an issue with these options? (call it "Consider optimizing history ...")